### PR TITLE
Pr create review tasklist for managers signing off cases

### PR DESCRIPTION
**Pull request by notapatch**
## Description of change

Create a tasklist page for the review phase of the project to contain the tasks needed to be undertaken as a review.

### User Story
As a reviewer I need to see all the tasks I need to complete so that I quickly review assessments and keep track of any changes needed

### ACs by commit
AC1: Link added to: Review and sign-off
AC2: Review assessment moved to Review and sign-off landing page
AC3: Reviewer can navigate back to application landing via back button
AC4: Is to avoid regression
AC5: Rename non-reviewers text to view recommendation

### Reviewer on the Application page - Review and sign-off

reviewer-review-and-sign-off

### Reviewer on the Review and sign-off landing page

reviewer-review-and-sign-off-landing-page

### Reviewer navigating from Review and sign-off landing page to Review the recommendation

reviewer-review-the-recommendation

### Assessor before determination
- No link and the text now reads 'View recommendation'
assessor-before-determination

### Assessor after determination

- Link as before reads 'View recommendation'

assessor-after-determination

### Story Link

https://trello.com/c/cuDX5hsQ/1263-create-review-tasklist-for-managers-signing-off-cases

### Known issues

Things you know need further follow on work but aren't in scope of this issue

1. Status tag is unlikely to be correct as it's the old status tag
2. Review the recommendation breadcrumbs are wrong
3. Assessor should be forbidden from the review tasklist landing page

### CODE REVIEW

File: app/views/planning_application/review_tasks/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<% content_for :page_title do %>
+  Review and sign-off - <%= t('page_title') %>
+<% end %>
+
+<% add_parent_breadcrumb_link "Home", planning_applications_path %>
+<% add_parent_breadcrumb_link "Application", planning_application_path(@planning_application) %>
+<% content_for :title, "Review and sign-off" %>
+
+<div class="govuk-grid-row">
```
Comment (notapatch) [inline app/views/planning_application/review_tasks/index.html.erb:9]: This code is taken from shared/_proposal_header.html.erb - they wanted to change what was displayed in the heading for a reviewer so I couldn't directly reuse. I wanted to create duplication until I knew that the designers really wanted and then refactor to eitehr create a new partial or to change shared/_proposal_header in some way.

**Review by EGiataganas — APPROVED** [review]

File: features/step_definitions/planning_application_steps.rb [hunk 2]
```diff
@@ -98,17 +98,6 @@
   )
 end
 
-Given("the planning application is determined") do
```
Comment (notapatch) [inline features/step_definitions/planning_application_steps.rb:101]: Removed cucumber with 'Review assessment' without breaking tests
  - the link 'Review assessment' is being changed so I was testing
    out the code but this didn't break anything so I've removed it

File: app/views/planning_applications/steps/_review.html.erb [hunk 3]
```diff
@@ -3,12 +3,18 @@
   <ul class="app-task-list__items">
     <li class="app-task-list__item">
       <span class="app-task-list__task-name">
-        <% if @planning_application.awaiting_determination? && current_user.reviewer? %>
-          <%= link_to "Review assessment", edit_planning_application_recommendations_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
-        <% elsif @planning_application.awaiting_determination? && !current_user.reviewer? %>
-          <%= link_to "View recommendation", view_recommendation_planning_application_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
+        <% if current_user.reviewer? %>
```
Comment (notapatch) [inline app/views/planning_applications/steps/_review.html.erb:6]: Refactor review logic into reviewer and !reviewer blocks

File: app/views/planning_applications/steps/_review.html.erb [hunk 4]
```diff
@@ -3,12 +3,18 @@
...
+            <%= link_to "Review and sign-off", planning_application_review_tasks_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
+          <% else %>
+            Review and sign-off
+          <% end %>
         <% else %>
-          Review assessment
+          <% if @planning_application.awaiting_determination? %>
+            <%= link_to "View recommendation", view_recommendation_planning_application_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
+          <% else %>
+            View recommendation
```
Comment (notapatch) [inline app/views/planning_applications/steps/_review.html.erb:16]: AC5: Rename non-reviewers text to view recommendation

File: app/views/planning_applications/steps/_review.html.erb [hunk 5]
```diff
@@ -3,12 +3,18 @@
   <ul class="app-task-list__items">
     <li class="app-task-list__item">
       <span class="app-task-list__task-name">
-        <% if @planning_application.awaiting_determination? && current_user.reviewer? %>
-          <%= link_to "Review assessment", edit_planning_application_recommendations_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
-        <% elsif @planning_application.awaiting_determination? && !current_user.reviewer? %>
-          <%= link_to "View recommendation", view_recommendation_planning_application_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
+        <% if current_user.reviewer? %>
+          <% if @planning_application.awaiting_determination? %>
+            <%= link_to "Review and sign-off", planning_application_review_tasks_path(@planning_application), aria: { describedby: "review_assessment-completed" }, class: "govuk-link" %>
```
Comment (notapatch) [inline app/views/planning_applications/steps/_review.html.erb:8]: Rename reviewers text to: Review and sign-off

File: app/views/recommendations/edit.html.erb [hunk 6]
```diff
@@ -2,8 +2,6 @@
   Review form - <%= t('page_title') %>
 <% end %>
 
-<% content_for :title, "Review assessment" %>
```
Comment (notapatch) [inline app/views/recommendations/edit.html.erb:5]: - Removed title as it wasn't being used
  - I'm having to change text which is 'Review assessment' but
    I don't see where this is used so I've removed it

File: app/views/planning_application/review_tasks/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    <h2 class="app-task-list__section">
+      Review assessment
+    </h2>
+    <ul class="app-task-list__items">
+      <li class="app-task-list__item">
+        <span class="app-task-list__task-name">
+          <%= link_to "Sign-off recommendation",
+                      edit_planning_application_recommendations_path(@planning_application),
+                      aria: { describedby: "review_assessment-completed" },
+                      class: "govuk-link" %>
```
Comment (notapatch) [inline app/views/planning_application/review_tasks/index.html.erb:50]: - AC2: Review assessment moved to Review and sign-off landing page

File: app/views/planning_application/review_tasks/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        </span>
+      </li>
+    </ul>
+  </div>
+</div>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <%= link_to "Back", planning_application_path(@planning_application), class: "govuk-button govuk-button--secondary" %>
```
Comment (notapatch) [inline app/views/planning_application/review_tasks/index.html.erb:59]: - AC3: Reviewer can navigate back to application landing via back button

**Pull request merged**